flink-table/flink-table-planner/src/test/scala/org/apache/flink/table/planner/runtime/harness/RankHarnessTest.scala

```diff
@@ -220,4 +221,74 @@ class RankHarnessTest(mode: StateBackendMode) extends HarnessTestBase(mode) {
     assertor.assertOutputEqualsSorted("result mismatch", expectedOutput, result)
     testHarness.close()
   }
+
+  @Test
+  def testUpdateRankWithRowNumber(): Unit = {
```

Comment:
Add more test scenarios.
1. Calculate top 5 but there are more than 5 candidates.
2. Sort key drops but ranking does not change.
3. Sort key drops but does not drop to the last ranking.
4. Calculate top 5, 7 candidates, previous rank 3 drops to rank 6 (but it is still "rank 5").

Reply:
Thanks for your review, I'll update it.